### ISSUE TITLE
Memory leak while trying to index

### ISSUE THREAD

**Issue by ging-dev** [phpactor/phpactor#2377]
Step to reproduce:
- Clone https://github.com/phan/phan
- Open project and `phpactor` stuck while indexing

**Comment by dantleech** [comment]
Works for me, can you provide any logs?

**Comment by ging-dev** [comment]
> Works for me: 3892 files, ~250MB used at end of indexing, can you provide any logs?

![image](https://github.com/phpactor/phpactor/assets/42226341/b911b323-2c40-4839-b6d7-5e110e1f4031)

in `vscode`, I'm stuck at `resolving code actions` and LSP leaks memory, `phpactor index:build` work normal

**Comment by dantleech** [comment]
which version of `phpactor` / `vscode-phpactor`  are you using? It should look like this:

![image](https://github.com/phpactor/phpactor/assets/530801/8176cb65-3b3e-4de9-9273-5cf48ea33dc0)

**Comment by ging-dev** [comment]
> which version of `phpactor` / `vscode-phpactor` are you using? It should look like this:
> 
> ![image](https://user-images.githubusercontent.com/530801/270130352-8176cb65-3b3e-4de9-9273-5cf48ea33dc0.png)

I use phpactor/vscode-phpactor latest version, I think it stuck when we open `src/Phan/Language/Type.php`

**Comment by dantleech** [comment]
still works fine for me :thinking: 

![image](https://github.com/phpactor/phpactor/assets/530801/fc4da56e-af8f-48d7-a50b-de70537f7d83)

(code actions resolve etc)

**Comment by ging-dev** [comment]
> still works fine for me 🤔
> 
> ![image](https://user-images.githubusercontent.com/530801/270135279-fc4da56e-af8f-48d7-a50b-de70537f7d83.png)
> 
> (code actions resolve etc)

When I checked the log, no unusual behavior was detected. Strangely, my process crashes and keeps running out of memory

When run `phpactor worse:analyse ./` give me error:
```
phpactor worse:analyse ./
Analysing files...

   40/2394 [>---------------------------]   1%
In Analyser.php line 43:                                                                                                                                                                                                                                                         
  Error while analysing file "/home/gingdev/Code/bugs/phan/.phan/plugins/PrintfCheckerPlugin.php": Phpactor\DocblockParser\Ast\Type\NullableNode::__construct(): Argument #2 ($type) must be of  
   type Phpactor\DocblockParser\Ast\TypeNode, null given, called in /home/gingdev/.config/Code/User/globalStorage/dantleech.vscode-phpactor/phpactor/lib/DocblockParser/Parser.php on line 256   
                                                                                                                                                                                                 

In NullableNode.php line 15:
                                                                                                                                                                                                 
  Phpactor\DocblockParser\Ast\Type\NullableNode::__construct(): Argument #2 ($type) must be of type Phpactor\DocblockParser\Ast\TypeNode, null given, called in /home/gingdev/.config/Code/User  
  /globalStorage/dantleech.vscode-phpactor/phpactor/lib/DocblockParser/Parser.php on line 256                                                                                                    
                                                                                                                                                                                                 

worse:analyse [--format FORMAT] [--ignore-failure] [--] <path>
```

```
phpactor worse:analyse src/Phan/Language/Type
Analysing files...

  0/56 [>---------------------------]   0%
In Analyser.php line 43:
                                                                                                                                                                                                 
  Error while analysing file "/home/gingdev/Code/bugs/phan/src/Phan/Language/Type/ArrayShapeType.php": Phpactor\DocblockParser\Ast\Type\NullableNode::__construct(): Argument #2 ($type) must b  
  e of type Phpactor\DocblockParser\Ast\TypeNode, null given, called in /home/gingdev/.config/Code/User/globalStorage/dantleech.vscode-phpactor/phpactor/lib/DocblockParser/Parser.php on line   
  256                                                                                                                                                                                            
                                                                                                                                                                                                 

In NullableNode.php line 15:
                                                                                                                                                                                                 
  Phpactor\DocblockParser\Ast\Type\NullableNode::__construct(): Argument #2 ($type) must be of type Phpactor\DocblockParser\Ast\TypeNode, null given, called in /home/gingdev/.config/Code/User  
  /globalStorage/dantleech.vscode-phpactor/phpactor/lib/DocblockParser/Parser.php on line 256                                                                                                    
                                                                                                                                                                                                 

worse:analyse [--format FORMAT] [--ignore-failure] [--] <path>
```

**Comment by dantleech** [comment]
The fatal error should be fixed in #2379 after which PHP Phpactor reports `mem: 468,669,152b` in this project on `UniontType.php` (the one with the docblock which caused the crash).
